### PR TITLE
Serialize/deseialize list of DTO using same mechanism as single DTO object

### PR DESCRIPTION
**Pull request by mshaposhnik**
### What does this PR do?
Different ways of serialization/deserialization of the single and list of `DTO` may provide erroneous effects, when, for example, Gson type adapters is applied when single DTO is requested, and not applied when list is requested, so the same objects have different representation. This PR corrects situation for  DTO lists.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/14739

#### Release Notes
N/A

#### Docs PR
N/A

### CODE REVIEW

**Review by skabashnyuk — APPROVED** [review]
Please wait for at least one more review

**Review by sleshchenko — APPROVED** [review]
Can not say that I reviewed very precisely but LGTM in general

**Pull request merged**
